site/content/docs/5.0/utilities/api.md

```diff
@@ -291,3 +291,59 @@ $utilities: (
   "float": null,
 );
 ```
+
+## Adding pseudo-classes to utilities
+
+With the `state` option, pseudo-classes utility classes can be generated:
+
+```scss
+$utilities: (
+  "shadow": (
+    property: box-shadow,
+    state: hover focus,
+    class: shadow,
+    values: (
+      null: $box-shadow,
+      sm: $box-shadow-sm,
+      lg: $box-shadow-lg,
+      none: none,
+    )
+  )
+);
+```
+
+Output:
+
+```css
+.shadow-hover:hover {
+  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15) !important;
+}
```

Comment:
Per my above comment, I think just `.shadow:hover` makes the most sense. Here we'd be allowing folks to opt-in to any state(s) for any classes.

```suggestion
.shadow:hover {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15) !important;
}
```

Reply:
А как в таком случае задать hover? Классы shadow и псевдокласс shadow:hover никак не будут отличаться. Это не приведёт ни к какому эффекту. Например: блок с shadow-sm при hover нужно изменить на shadow-lg

```html
<div class="shadow-sm"></div>
```

```css
.shadow-sm:hover {
  /* ??? He will repeat his own styles */
}
```

Reply:
And it should look like this:

```html
<div class="shadow-sm shadow-lg-hover"></div>
```

or 

```html
<div class="shadow-sm shadow-hover-lg"></div>
```

Reply:
> Per my above comment, I think just `.shadow:hover` makes the most sense. Here we'd be allowing folks to opt-in to any state(s) for any classes.

This confuses me a bit @mdo. This means adding the `shadow` class would only add a shadow on hover?

Reply:
Hmm might have been a brain fart from me—for some reason I was thinking of these as adding to the existing utilities, but that's not right as y'all have mentioned. So, my apologies and carry on!

Reply:
I would suggest generating with this pattern

```
.#{$pseudo}-#{$property-class + $infix + $property-class-modifier}:#{$pseudo}
```

This will separate the utility and its states. The user will be able to work faster with pseudo-classes.

For instance:

```
.hover-shadow-sm
.hover-text-primary
.focus-border-info
```

etc

Reply:
Problem with that is it breaks our conventions elsewhere. I think we'd have to do the state at the end. Thoughts @MartijnCuppens and @ffoodd?

Reply:
@mdo Then I propose to leave it as it was. Classes .shadow-sm-hover etc